### DISCUSSION TITLE
Textpath scale handling

### DISCUSSION THREAD

**Discussion by melchiar** [fabricjs/fabric.js#7161] [Ideas]
Text on a path currently ignores the path's scale. Resolving this limitation raises a couple of questions.

![image](https://user-images.githubusercontent.com/9057412/123467534-77bb0f00-d5a5-11eb-852a-1b6f5a71f5b8.png)

1. Ideally, control resizing of a text object on a path would transform the path itself rather than the text. Should this be left to the application, or done natively?

2. The scale handling logic itself is going to be tricky to implement since scale will interfere with both angles and distances. Should we take the approach of transforming the entire path coordinates first, or just account for scale in the `getPointOnPath` method?

**Comment by asturur** [comment]
So the whole path semplification has been done because you can also now transform paths point.
In my opinion this is application businnes.
You get custom control api, and with the controls you decide if to scale the object ( default ) or scale the inner path.
so you calculate the next scale, you apply to the path, you change the path and you recalculate text drawing.

I think we can demo it already with the actual code.

**Reply by melchiar** [reply]
Sounds good, yes scaling of the path is simple enough to do with the new controls api. 

Don't you think that a path's scale should have an effect on the text automatically though? Currently text just ignores the scale applied to the path which seems likely to become a point of confusion.

**Reply by asturur** [reply]
I do not mean to say path.scaleX = something, but instead to take each path point and apply a scale transform to it. That would give you a new path that unscaled, looks like the other scaled.
At that point the text will not ignore it.
